core/templates/pages/admin-page/activities-tab/admin-dev-mode-activities-tab.component.spec.ts

```diff
@@ -0,0 +1,415 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { FormsModule } from '@angular/forms';
+import { MaterialModule } from 'components/material.module';
+
+import { AdminBackendApiService, AdminPageData } from 'domain/admin/admin-backend-api.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+import { AdminDataService } from '../services/admin-data.service';
+import { AdminTaskManagerService } from '../services/admin-task-manager.service';
+import { AdminDevModeActivitiesTabComponent } from './admin-dev-mode-activities-tab.component';
+
+describe('Admin dev mode activities tab', () => {
+  let component: AdminDevModeActivitiesTabComponent;
+  let fixture: ComponentFixture<AdminDevModeActivitiesTabComponent>;
+  let adminBackendApiService: AdminBackendApiService;
+  let adminDataService: AdminDataService;
+  let adminTaskManagerService: AdminTaskManagerService;
+  let windowRef: WindowRef;
+  let adminDataObject = {
+    demoExplorationIds: ['expId'],
+    demoExplorations: [
+      [
+        '0',
+        'welcome.yaml'
+      ]
+    ],
+    demoCollections: [
+      ['collectionId']
+    ],
+  } as AdminPageData;
+  let mockConfirmResult: (val: boolean) => void;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        MaterialModule,
+        FormsModule,
+        HttpClientTestingModule
+      ],
+      declarations: [
+        AdminDevModeActivitiesTabComponent
+      ]
+    }).compileComponents();
+  }));
+
+  beforeEach(async(() => {
+    fixture = TestBed.createComponent(AdminDevModeActivitiesTabComponent);
+    component = fixture.componentInstance;
+    adminBackendApiService = TestBed.get(AdminBackendApiService);
+    adminDataService = TestBed.get(AdminDataService);
+    adminTaskManagerService = TestBed.get(AdminTaskManagerService);
+    windowRef = TestBed.get(WindowRef);
+
+    spyOn(adminDataService, 'getDataAsync').and.resolveTo(adminDataObject);
+
+    let confirmResult = true;
+    spyOnProperty(windowRef, 'nativeWindow').and.returnValue({
+      confirm: () => confirmResult
+    });
+
+    mockConfirmResult = (val) => {
+      confirmResult = val;
+    };
+
+    fixture.detectChanges();
+    component.ngOnInit();
+  }));
+
+  it('should create', () => {
+    expect(component).toBeDefined();
+  });
+
+  it('should initialize data correctly', () => {
+    expect(component.reloadingAllExplorationPossible).toEqual(true);
+    expect(component.demoExplorationIds).toEqual(['expId']);
+    expect(component.DEMO_COLLECTIONS).toEqual([['collectionId']]);
+    expect(component.DEMO_EXPLORATIONS).toEqual([['0', 'welcome.yaml']]);
+  });
+
+  describe('.reloadExploration', () => {
+    it('should not reload a specific exploration if task is running', () => {
+      spyOn(
+        adminTaskManagerService, 'isTaskRunning').and.returnValue(true);
+      expect(component.reloadExploration('expId')).toBeUndefined();
```

Comment:
Spy on `.adminBackendApiService.reloadExplorationAsync` and assert this function didn't get called*

Reply:
Did you mean to change the ```it``` description or the test.

I added the test to check if the backend api service was not called. Pl ping if you want the description to be changed too,

Reply:
> I added the test to check if the backend api service was not called. Pl ping if you want the description to be changed too,

The description is fine, it matches the assertion of not calling reload function.